### PR TITLE
feat: redux login implementation

### PR DESCRIPTION
**Pull request by harlyon**
Overview 🚀
Implemented Redux login.

Type of change 🧙‍
🎉 Initial commit (area): a brief description of change.

✨ New feature (area): a brief description of change.

🐛 Bugfix (area): a brief description of change.

📚 Documentation (area): a brief description of change.

🐎 Performance (area): a brief description of change.

🚨 Tests (area): a brief description of change.

💚 Continuous Integration (area): a brief description of change.

🔒 Security (area): a brief description of change.

👕 Lint (area): a brief description of change.

🚀 Deploying stuff (area): a brief description of change.

🚧 Work in progress (area): a brief description of change.

🔧 Configuration files (area): a brief description of change.

💩 Bad code / need improv. (area): a brief description of change.

💥 Breaking changes (area): a brief description of change.

👌 Code review changes (area): a brief description of change.

How Has This Been Tested? 🚨
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

✅ Test A

❌ Test B

Screenshots 📸
If you changed the UI or UX include a screenshot, video or gif

Checklist ✅
[ ] My code follows the style guidelines of this project

[ ] I have performed a self-review of my own code

[ ] I have commented my code, particularly in hard-to-understand areas

[ ] I have made corresponding changes to the documentation

[ ] My changes generate no new warnings

[ ] I have added tests that prove my fix is effective or that my feature works

[ ] New and existing unit tests pass locally with my changes

[ ] Any dependent changes have been merged & published in downstream modules

### CODE REVIEW

File: src/redux/actions/authAction.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  }
+};
+
+
+//login success
+export const login=(user)=>async dispatch=>{
+  const config={
+    headers: { 'Content-Type': 'application/json'}
+  }
+try {
```
Comment (yoniakabecky) [inline src/redux/actions/authAction.js:51]: Better to align indent 🤔

File: src/redux/actions/authAction.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+//load user
+export const loadUser=()=>async (dispatch)=>{
+
+  //set global headers
+  if(localStorage.token){
+    setAuthToken(localStorage.token)
+  }
+
+try {
```
Comment (yoniakabecky) [inline src/redux/actions/authAction.js:80]: Same here

File: src/redux/reducer/authReducer.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  loading: false,
+  error: null,
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case REGISTER_SUCCESS:
+    case LOGIN_SUCCESS:
+      localStorage.setItem('token', action.payload.token);
+      console.log(action.payload)
```
Comment (yoniakabecky) [inline src/redux/reducer/authReducer.js:25]: is this console.log necessary?

Comment (harlyon) [reply src/redux/reducer/authReducer.js]: fixed

File: src/redux/actions/authAction.js (hunk 2 above)
Comment (harlyon) [reply src/redux/actions/authAction.js]: added prettier to cater for this

File: src/redux/actions/authAction.js (hunk 1 above)
Comment (harlyon) [reply src/redux/actions/authAction.js]: added prettier to cater for this

**Review by yoniakabecky — APPROVED** [review]
LGTM 😊

**Pull request merged**